runtime/src/consensus/tendermint/verifier/store/lru.rs

```diff
@@ -94,12 +85,22 @@ impl LightStore for LruStore {
             store.lowest_height = Some(height);
```

Comment:
Lowest height should be renamed to lowest trusted and promoted always except when the newly added is trusted+lower. In that case the update is needed.

Maybe we simplify this by adding `trust_root_height` to the constructor and always promote it, even if it was somehow removed. We need to revert `lowest` method in that case to previous version.
```
// Promote trust root to prevent it from being evicted.
if let Some(trust_root_height) = store.trust_root_height {
    store.blocks.promote(&trust_root_height);
}
```